src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/col2im.cpp

```diff
@@ -0,0 +1,327 @@
+        result << "padsEnd=" << pads_end << "_";
+        result << "dataPrecision=" << netPrecision << "_";
+        result << "indexPrecision=" << indexPrecision;
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+
+        return result.str();
+    }
+protected:
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        ov::Tensor data_tensor;
+        const auto& dataPrecision = funcInputs[0].get_element_type();
+        const auto& dataShape = targetInputStaticShapes.front();
+        ov::test::utils::InputGenerateData in_data;
+        in_data.start_from = 0;
+        in_data.range = 10;
+        in_data.resolution = 1000;
+        data_tensor = ov::test::utils::create_and_fill_tensor(dataPrecision, dataShape, in_data);
+        inputs.insert({ funcInputs[0].get_node_shared_ptr(), data_tensor });
+    }
+
+    void SetUp() override {
+        Col2ImLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        Col2ImSpecificParams Col2ImParams;
+        ElementType inputPrecision;
+        ElementType indexPrecision;
+        std::tie(Col2ImParams, inputPrecision, indexPrecision, targetDevice) = basicParamsSet;
+
+        InputShape dataInputShape;
+        std::vector<int64_t> outputSize;
+        std::vector<int64_t> kernelSize;
+        ov::Strides strides;
+        ov::Strides dilations;
+        ov::Shape pads_begin;
+        ov::Shape pads_end;
+        std::tie(dataInputShape, outputSize, kernelSize, strides, dilations, pads_begin, pads_end) = Col2ImParams;
+
+        auto image2DSpatialShape = InputShape{{}, {{2}}};
+        init_input_shapes({dataInputShape, image2DSpatialShape, image2DSpatialShape});
+        auto dataParameter = std::make_shared<ov::op::v0::Parameter>(inputPrecision, inputDynamicShapes[0]);
+        auto outputSizeConst = std::make_shared<ov::op::v0::Constant>(indexPrecision, ov::Shape{ 2 }, outputSize);
+        auto kernelSizeConst = std::make_shared<ov::op::v0::Constant>(indexPrecision, ov::Shape{ 2 }, kernelSize);
+
+        auto Col2Im = std::make_shared<ov::op::v15::Col2Im>(dataParameter,
+                                                            outputSizeConst,
+                                                            kernelSizeConst,
+                                                            strides,
+                                                            dilations,
+                                                            pads_begin,
+                                                            pads_end);
+
+        ov::ParameterVector params{ dataParameter };
+        function = makeNgraphFunction(inputPrecision, params, Col2Im, "Col2Im");
+
+        if (inputPrecision == ov::element::bf16) {
```

Comment:
Better to add hardware check `inputPrecision == ov::element::bf16 && !ov::intel_cpu::hasHardwareSupport(inputPrecision)`

Reply:
Discussed offline to continue in 145726.